src/test/java/com/ss/training/utopia/agent/ServiceTestSuite.java

```diff
@@ -96,19 +96,53 @@ public void cancelBooking() {
         Booking cancellableBooking = new Booking(oneLong, oneLong, oneLong, true, null);
 
         flightDAO.save(futureFlight);
-        bookingDao.save(cancellableBooking);
+        bookingDAO.save(cancellableBooking);
 
         List<Booking> foundBookings;
-        foundBookings = bookingDao.findCancellable(oneLong);
+        foundBookings = bookingDAO.findCancellable(oneLong);
 		assertEquals(foundBookings.size(), 1);
 
         service.cancelBooking(cancellableBooking);
         
-        Booking cancelledBooking = bookingDao.findAll().get(0);
+        Booking cancelledBooking = bookingDAO.findAll().get(0);
         assertEquals(cancelledBooking.getActive(), false);
 
 
-        foundBookings = bookingDao.findCancellable(oneLong);
+        foundBookings = bookingDAO.findCancellable(oneLong);
 		assertEquals(foundBookings.size(), 0);
     }
+
+    @Test
+	public void readAvailableFlights() {
+       //Longs
+       Long oneLong = (long) 1;
+       Long twoLong = (long) 2;
+       Long threeLong = (long) 3;
+
+       // Times
+       final Long HOUR = (long) 3_600_000;
+       Long now = Instant.now().toEpochMilli();
+       Timestamp past = new Timestamp(now - HOUR);
+       Timestamp future = new Timestamp(now + HOUR);
+
+       Flight futureFlight = new Flight(oneLong, twoLong, future, oneLong, null, null);
+       Flight pastFlight = new Flight(oneLong, twoLong, past, twoLong, null, null);
+       Flight otherFutureFlight = new Flight(twoLong, oneLong, future, threeLong, null, null);
+
+       flightDAO.save(futureFlight);
+       flightDAO.save(pastFlight);
+       flightDAO.save(otherFutureFlight);
+
+       List<Flight> flights = service.readAvailableFlights();
+
+        assertEquals(flights.size(), 2);
+    }
```

Comment:
For this test it would be good to make sure the flights that get found are the right ones, and also to make sure it only finds flights with available seats.